src/Corax/Querying/TermsReader.cs

```diff
@@ -41,6 +51,79 @@ public string GetTermFor(long id)
         return s;
     }
     
+    /// <summary>
+    /// Read terms in bulk. The result is not associated with ids by index.
+    /// Caution:
+    /// - the `termsSet` span is being invalidated between calls.
+    /// - the method modifies 'ids' (change the order) of the buffer (!)
+    /// - the order of the terms is based on sorted ids values
+    /// </summary>
+    public int GetAllTermsFromSet(Span<long> ids, out Span<UnmanagedSpan> termsSet)
+    {
+        const int pageSizeShift = 13;
+        Debug.Assert(1 << pageSizeShift == (long)Voron.Global.Constants.Storage.PageSize);
+        var maxToProcess = Math.Min(ids.Length, 1024);
```

Comment:
Is there a reason why you put a limit here? If so, we probably should document here with a comment

Reply:
> I wonder whether we shouldn't do it outside? So the called would handle the batching explicitly

The TermsReader is the cache owner, we do not want upper caller to reserve big chunks. The reader should take care of this instead.

Reply:
Done